crates/loader/src/cache.rs

```diff
@@ -17,6 +17,7 @@ const DATA_DIR: &str = "data";
 pub struct Cache {
     /// Root directory for the cache instance.
     root: PathBuf,
+    dirs_ensured: tokio::sync::RwLock<bool>,
```

Comment:
Can we use `AtomicBool` instead?

And nit: please add a comment for what this flag is used for

Reply:
I also like `dirs_ensured_once: tokio::sync::OnceCell<()>`:

https://docs.rs/tokio/1.36.0/tokio/sync/struct.OnceCell.html

```rust
self.dirs_ensured_once.get_or_init(|| async {
    // ensure yer dirs
}).await
```